GH-3844: Rework messaging annotation with @Bean (#3877)
spring-projects/spring-integration: src/reference/asciidoc/configuration.adoc

* The `MessageHandler` (`MessageSource`) `@Bean` gets its own standard name from the method name or `name` attribute on the `@Bean`.
This works as though there were no messaging annotation on the `@Bean` method.
* The `AbstractEndpoint` bean name is generated with the following pattern: `[@Bean name].[decapitalizedAnnotationClassShortName]`.
For example, the `SourcePollingChannelAdapter` endpoint for the `consoleSource()` definition <<annotations_on_beans,shown earlier>> gets a bean name of `consoleSource.inboundChannelAdapter`.

> Unlike with POJO methods the bean method name is not included into an endpoint bean name.

Reviewer: ```suggestion
Unlike with POJO methods, the bean method name is not included in the endpoint bean name.
```
